src/hardware/vga_s3.cpp

```diff
@@ -902,24 +902,19 @@ struct PCI_VGADevice : public PCI_Device {
 		//registers[0x3c] = 0x0b;	// irq line
 		//registers[0x3d] = 0x01;	// irq pin
 
-		// uint32_t gfx_address_space=(((uint32_t)S3_LFB_BASE)&0xfffffff0)
-		// | 0x08;	// memory space, within first 4GB, prefetchable
-		uint32_t gfx_address_space = (((uint32_t)S3_LFB_BASE) &
-		                              0xfffffff0); // memory space,
-		                                           // within first 4GB
-		registers[0x10] = (uint8_t)(gfx_address_space & 0xff); // base
-		                                                       // addres 0
-		registers[0x11] = (uint8_t)((gfx_address_space >> 8) & 0xff);
-		registers[0x12] = (uint8_t)((gfx_address_space >> 16) & 0xff);
-		registers[0x13] = (uint8_t)((gfx_address_space >> 24) & 0xff);
-
-		uint32_t gfx_address_space_mmio = (((uint32_t)S3_LFB_BASE + 0x1000000) &
-		                                   0xfffffff0); // memory space,
-		                                                // within first 4GB
-		registers[0x14] = (uint8_t)(gfx_address_space_mmio & 0xff); // base addres 0
-		registers[0x15] = (uint8_t)((gfx_address_space_mmio >> 8) & 0xff);
-		registers[0x16] = (uint8_t)((gfx_address_space_mmio >> 16) & 0xff);
-		registers[0x17] = (uint8_t)((gfx_address_space_mmio >> 24) & 0xff);
+		// BAR0 - memory space, within first 4GB
+		static_assert((PCI_GFX_LFB_BASE & 0xf) == 0);
```

Comment:
Maybe a commenting that you're asserting 16-byte alignment here.

Reply:
OK, added a comment - in every such place.